Documents/Command/ConfigureAgentLogsUploadToCloudwatch/ConfigureAgentLogsUploadToCloudwatch.json

```diff
@@ -1,257 +1,326 @@
 {
-    "schemaVersion":"2.2",
-    "description":"Enable or Disable Pushing Agent Logs To CloudWatchLogs For Diagnostic Purposes",
-    "parameters": {
-      "LogLevel": {
-        "type": "String",
-        "description": "Log Level of SSM Agent",
-        "allowedValues": [
-          "Debug",
-          "Info",
-          "Error"
-        ]
+  "schemaVersion":"2.2",
+  "description":"Enable or Disable Pushing Agent Logs To CloudWatchLogs For Diagnostic Purposes",
+  "parameters": {
+    "LogLevel": {
+      "type": "String",
+      "description": "Log Level of SSM Agent",
+      "allowedValues": [
+        "Debug",
+        "Info",
+        "Error"
+      ]
+    },
+    "EnableCloudWatchUpload": {
+      "type": "String",
+      "description": "Enable Upload of Agent Logs To CloudWatch",
+      "allowedValues": [
+        "true",
+        "false"
+      ]
+    },
+    "LogGroup": {
+      "type": "String",
+      "default": "",
+      "description": "(Optional) Log Group Name. Will use the default group name if not specified. The log group would be created if not already present."
+    },
+    "EnableShare": {
+      "type": "String",
+      "default": "false",
+      "description": "(Optional) Enable sharing of logs with CloudWatch of another AWS Account. The ShareDestination parameter needs to be passed if true.",
+      "allowedValues": [
+        "true",
+        "false"
+      ]
+    },
+    "ShareDestination": {
+      "type": "String",
+      "default": "",
+      "description": "(Optional) Destination in the format 'IAMUserAccessKeyID::IAMUserSecretAccessKey::DestinationLogGroup::DestinationLogStream'. The IAM User should have access to DescribeLogStreams, PutLogEvents on the destination CloudWatchLogs resource. The log group and stream should be present. Parameter will be ignored if EnableShare is false",
```

Comment:
We should never pass secrets directly as input parameters.
Use Parameter Store and a SecureString parameter.

Reply:
Is it better to set default of ShareDestination to parameter from parameter store, or simply fetch it in PowerShell and shell script?

Reply:
The default would be to not share the logs with another account, so you can't set it to a specific parameter name.
I would keep the default value as is (blank), or set it to a placeholder string of your choosing (such as a "DO_NOT_SHARE" string value).
If you use a placeholder string, it's a little easier to implement a flow where you check any passed Parameter is a SecureString before it reaches your Powershell/Bash script. For example:
- Use [aws:branch](https://docs.aws.amazon.com/systems-manager/latest/userguide/automation-branchdocs.html) on the value of {{ ShareDestination }}
  - If {{ ShareDestination }} is not the default placeholder value (i.e. DO_NOT_SHARE), then jump to an aws:assertAwsResourceProperty step, where you want to assert the parameter is of type SecureString. You can call [this API](https://docs.aws.amazon.com/cli/latest/reference/ssm/get-parameter.html) to assert that
  - In all other cases, just continue with the automation without the SecureString check.